### PR TITLE
-Added Synthesis (resources analysis) and a new LRU replacement policy (LRU_stack, still in testing)

### PR DESCRIPTION
**Pull request by joaovroque**
- Added hardware/fpga/ in order to synthetize the cache (or other module, just change the top's name in synth.tcl): an easier way to analyze the resources, specially for remote work (opening the GUI is an impossibility (time-wise)).
- Added another approach to the LRU algorithm (`LRU_stack), which uses shifts instead of counters, simulating a Stack like priority (where the MRU is the Heap and LRU is the Tail). Unfortunately the performance is either the same or worse, the simulation of the iob-cache with it fails, despise working in the rp simulation. - Do not use this, currently.

### CODE REVIEW

File: hardware/fpga/AES-KU040-DB-G/synth.tcl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    if {$file != ""} {
+        read_verilog -sv $file
+    }
+}
+
+#read_verilog $VHDR
+
+#read_verilog ../../src/iob-cache.v
+read_verilog ../../include/iob-cache.vh
+#read_verilog ../../../submodule/iob-mem/reg_file/iob_reg_file.v
```
Comment (jjts) [inline hardware/fpga/AES-KU040-DB-G/synth.tcl:24]: delete useless comments

File: hardware/fpga/AES-KU040-DB-G/synth.tcl [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+foreach file [split $VSRC \ ] {
+    if {$file != ""} {
+        read_verilog -sv $file
+    }
+}
+
+#read_verilog $VHDR
+
+#read_verilog ../../src/iob-cache.v
+read_verilog ../../include/iob-cache.vh
```
Comment (jjts) [inline hardware/fpga/AES-KU040-DB-G/synth.tcl:23]: perhaps there's an include search path

File: hardware/simulation/icarus/Makefile [hunk 3]
```diff
@@ -1,8 +1,8 @@
 #module paths
-CACHE_DIR:=../../..
-INTERCON_DIR:=$(CACHE_DIR)/submodules/iob-interconnect
-MEM_DIR:=$(CACHE_DIR)/submodules/iob-mem
-AXI_MEM_DIR:=$(CACHE_DIR)/submodules/axi-mem
+ROOT_DIR:=../../..
+INTERCON_DIR:=$(ROOT_DIR)/submodules/iob-interconnect
```
Comment (jjts) [inline hardware/simulation/icarus/Makefile:3]: useless repetition, see?

Comment (joaovroque) [reply hardware/simulation/icarus/Makefile]: Sure thing, everything fixed now. Sorry for the trouble.

**Comment by jjts** [comment]
Unfortunately Github can't show me what you fixed . I will close this PR and ask you to open  new one.

**Pull request closed without merging**